app/src/main/java/de/westnordost/streetcomplete/quests/building_type/BuildingType.kt

```diff
@@ -0,0 +1,76 @@
+    APARTMENTS    ("building=apartments"),
+    DETACHED      ("building=detached"),
+    SEMI_DETACHED ("building=semidetached_house"),
+    TERRACE       ("building=terrace"),
+    HOTEL         ("building=hotel"),
+    DORMITORY     ("building=dormitory"),
+    HOUSEBOAT     ("building=houseboat"),
+    BUNGALOW      ("building=bungalow"),
+    STATIC_CARAVAN("building=static_caravan"),
+    HUT           ("building=hut"),
+
+    INDUSTRIAL    ("building=industrial"),
+    RETAIL        ("building=retail"),
+    OFFICE        ("building=office"),
+    WAREHOUSE     ("building=warehouse"),
+    KIOSK         ("building=kiosk"),
+    STORAGE_TANK  ("man_made=storage_tank"),
+
+    KINDERGARTEN  ("building=kindergarten"),
+    SCHOOL        ("building=school"),
+    COLLEGE       ("building=college"),
+    SPORTS_CENTRE ("building=sports_centre"),
+    HOSPITAL      ("building=hospital"),
+    STADIUM       ("building=stadium"),
+    TRAIN_STATION ("building=train_station"),
+    TRANSPORTATION("building=transportation"),
+    UNIVERSITY    ("building=university"),
+    GOVERNMENT    ("building=government"),
+
+    CHURCH        ("building=church"),
+    CHAPEL        ("building=chapel"),
+    CATHEDRAL     ("building=cathedral"),
+    MOSQUE        ("building=mosque"),
+    TEMPLE        ("building=temple"),
+    PAGODA        ("building=pagoda"),
+    SYNAGOGUE     ("building=synagogue"),
+    SHRINE        ("building=shrine"),
+
+    CARPORT       ("building=carport"),
+    GARAGE        ("building=garage"),
+    GARAGES       ("building=garages"),
+    PARKING       ("building=parking"),
+
+    FARM          ("building=farm"),
+    FARM_AUXILIARY("building=farm_auxiliary"),
+    SILO          ("man_made=silo"),
+    GREENHOUSE    ("building=greenhouse"),
+
+    SHED          ("building=shed"),
+    ROOF          ("building=roof"),
+    TOILETS       ("building=toilets"),
+    SERVICE       ("building=service"),
+    HANGAR        ("building=hangar"),
+    BUNKER        ("building=bunker"),
+    HISTORIC      ("historic=yes"),
+    ABANDONED     ("abandoned=yes"),
+    RUINS         ("ruins=yes"),
+
+    RESIDENTIAL   ("building=residential", listOf(DETACHED, APARTMENTS, SEMI_DETACHED, TERRACE, HOUSE, FARM, HUT, BUNGALOW, HOUSEBOAT, STATIC_CARAVAN, DORMITORY)),
+    COMMERCIAL    ("building=commercial", listOf(OFFICE, INDUSTRIAL, RETAIL, WAREHOUSE, KIOSK, HOTEL, STORAGE_TANK)),
+    CIVIC         ("building=civic", listOf(SCHOOL, UNIVERSITY, HOSPITAL, KINDERGARTEN, SPORTS_CENTRE, TRAIN_STATION, TRANSPORTATION, COLLEGE, GOVERNMENT, STADIUM)),
+    RELIGIOUS     ("building=religious", listOf(CHURCH, CATHEDRAL, CHAPEL, MOSQUE, TEMPLE, PAGODA, SYNAGOGUE, SHRINE)),
+    FOR_CARS      (null, listOf(GARAGE, GARAGES, CARPORT, PARKING)),
+    FOR_FARMS     (null, listOf(FARM, FARM_AUXILIARY, SILO, GREENHOUSE, STORAGE_TANK)),
+    OTHER         (null, listOf(SHED, ROOF, SERVICE, HUT, TOILETS, HANGAR, BUNKER, HISTORIC, ABANDONED, RUINS));
```

Comment:
This will probably not work like that unless `fun BuildingType.asItem()` works recursively.

Reply:
FOR_CARS, FOR_FARMS etc. are rather an UI category, so question is whether they should be part of the data at all or not just part of the UI data

Reply:
Indeed, that's a regression in [`644ee41` (#2495)](https://github.com/streetcomplete/StreetComplete/pull/2495/commits/644ee41b196beb7eebd2bc444a7d4e3e105a496c). I'll see what I can do instead.